x/superfluid/keeper/stake_test.go

```diff
@@ -663,334 +659,244 @@ func (suite *KeeperTestSuite) TestSuperfluidUnbondLock() {
-// 			intermediaryDels := []sdk.Dec{}
+					targetDelegation := intermediaryDels[index].Mul(multiplier2ByDenom[expAcc.Denom]).Quo(originMultiplier)
 
-// 			for _, intAccIndex := range tc.checkAccIndexes {
-// 				expAcc := intermediaryAccs[intAccIndex]
-// 				valAddr, err := sdk.ValAddressFromBech32(expAcc.ValAddr)
-// 				suite.Require().NoError(err)
+					// check delegation changes
+					delegation, found := suite.App.StakingKeeper.GetDelegation(suite.Ctx, expAcc.GetAccAddress(), valAddr)
 
-// 				// check delegation from intermediary account to validator
-// 				delegation, found := suite.App.StakingKeeper.GetDelegation(suite.Ctx, expAcc.GetAccAddress(), valAddr)
-// 				suite.Require().True(found)
-// 				intermediaryDels = append(intermediaryDels, delegation.Shares)
-// 			}
+					suite.Require().True(found)
+					suite.Require().Equal(delegation.Shares, targetDelegation)
+				}
+				return
+			}
 
-// 			for _, osmoEquivilentMultiplier := range tc.osmoEquivalentMultipliers {
-// 				suite.App.SuperfluidKeeper.SetOsmoEquivalentMultiplier(suite.Ctx, 2, osmoEquivilentMultiplier.denom, osmoEquivilentMultiplier.price)
-// 			}
-// 			suite.App.EpochsKeeper.SetEpochInfo(suite.Ctx, epochstypes.EpochInfo{
-// 				Identifier:   params.RefreshEpochIdentifier,
-// 				CurrentEpoch: 2,
-// 			})
-
-// 			suite.NotPanics(func() {
-// 				suite.App.SuperfluidKeeper.RefreshIntermediaryDelegationAmounts(suite.Ctx)
-// 			})
-
-// 			for index, intAccIndex := range tc.checkAccIndexes {
-// 				expAcc := intermediaryAccs[intAccIndex]
-// 				valAddr, err := sdk.ValAddressFromBech32(expAcc.ValAddr)
-// 				suite.Require().NoError(err)
+			// unbond all lockups
+			for _, lock := range locks {
+				// superfluid undelegate
+				// handling the case same lockup is used for further delegation
+				cacheCtx, write := suite.Ctx.CacheContext()
+				err := suite.App.SuperfluidKeeper.SuperfluidUndelegate(cacheCtx, lock.Owner, lock.ID)
+				if err == nil {
+					write()
+				}
+			}
+			unbondingDuration := suite.App.StakingKeeper.GetParams(suite.Ctx).UnbondingTime
 
-// 				// get the updated delegtaion
-// 				delegation, found := suite.App.StakingKeeper.GetDelegation(suite.Ctx, expAcc.GetAccAddress(), valAddr)
+			// check intermediary account changes after unbonding operations
+			for _, intAccIndex := range tc.checkAccIndexes {
+				expAcc := intermediaryAccs[intAccIndex]
+				suite.Ctx = suite.Ctx.WithBlockTime(suite.Ctx.BlockTime().Add(unbondingDuration + time.Second))
+				suite.App.EndBlocker(suite.Ctx, abci.RequestEndBlock{Height: suite.Ctx.BlockHeight()})
 
-// 			}
+				unbonded := suite.App.BankKeeper.GetBalance(suite.Ctx, expAcc.GetAccAddress(), sdk.DefaultBondDenom)
+				suite.Require().True(unbonded.IsZero())
+			}
 
-// 			panic("e")
-// 		})
-// 	}
-// }
+			// refresh intermediary account delegations
+			suite.NotPanics(func() {
+				suite.App.SuperfluidKeeper.RefreshIntermediaryDelegationAmounts(suite.Ctx)
+			})
```

Comment:
How would this panic? Isn't the sensitive logic all wrapped in an `ApplyIfNoErr` function?

Reply:
This won't panic - previously there was a case for panics - I will remove it